src/interpreter/executors/main.rs

```diff
@@ -85,34 +90,36 @@ where
 {
     // TODO maybe infallible?
```

Comment:
We could fail due to I/O issues or problems that trigger a `halt_on_bug`, so I don't think this TODO is needed.

```suggestion
```

Reply:
This was created a long time ago, when I was centralizing the I/O exceptions into `std::io::Error`. Its not really related to this PR, but won't hurt to remove it here

https://github.com/FuelLabs/fuel-vm/pull/173/commits/26b3663740c1064a3baa47434d2355d9f90c6289